Sources/HttpPipeline/Middleware.swift

```diff
@@ -47,26 +47,26 @@ public func end(conn: Conn<BodyOpen, Data?>) -> Conn<ResponseEnded, Data?> {
   )
 }
 
-public func end<A>(conn: Conn<HeadersOpen, A>) -> Conn<ResponseEnded, Data?> {
+public func end<A>(conn: Conn<HeadersOpen, A>) -> IO<Conn<ResponseEnded, Data?>> {
   return conn
     |> closeHeaders
-    |> map(const(nil))
-    |> end
+    >-> map(const(nil)) >>> pure
```

Comment:
middlewares now compose with kleisli `>->`, but middleware transformers will still compose with regular `<<<`

Reply:
oh and anything that is pure (like the `map(const(nil))` above) needs to be post composed with a `>>> pure`.